packages/frontend/app/pods/components/pod/application/top-nav/template.hbs

```diff
@@ -2,14 +2,14 @@
 
   <div class='left-nav'>
     {{#if this.currentUser.isLoggedIn}}
-      <div>
+      <div {{update-document-title unreadMessageCount=this.unreadMessageCount currentUserName=this.currentUser.name}}>
```

Comment:
I like where you're head's at with the readability here! it's very clear what argument is being used for what.
however, the currentUser doesn't need to be passed, because we can access it directly from the modifier, like so:

```ts
import { inject as service } from '@ember/service';

// ...

export default class ... extends ...  {
  @service currentUser;

   // ...

   document.title = `${unreadCount} ${this.currentUser.name} .... `;
}
```

and given that, we would then only need one argument passed to update-document-title, and since there is only one, it could be a positional param without any lost semantics.

thoughts?

Reply:
That is good for this current use. If there are future use cases that need to change the document title, it will need to be modified.

Reply:
What other uses can you think of?

Over designing up front is common, and often leads to incorrect abstractions ;)